FirebaseSessions/Tests/Unit/SessionGeneratorTests.swift

```diff
@@ -49,30 +76,29 @@ class IdentifiersTests: XCTestCase {
   // with the Sessions SDK, we may want to move to a lazy solution where
   // sessionID can never be empty
   func test_sessionID_beforeGenerateReturnsNothing() throws {
-    XCTAssert(identifiers.sessionID.count == 0)
-    XCTAssertNil(identifiers.previousSessionID)
+    XCTAssertNil(generator.currentSession())
   }
 
   func test_generateNewSessionID_generatesValidID() throws {
-    identifiers.generateNewSessionID()
-    XCTAssert(isValidSessionID(identifiers.sessionID))
-    XCTAssertNil(identifiers.previousSessionID)
+    let sessionInfo = generator.generateNewSession()
+    XCTAssert(isValidSessionID(sessionInfo.sessionId))
+    XCTAssertNil(sessionInfo.previousSessionId)
   }
 
   /// Ensures that generating a Session ID multiple times results in the last Session ID being set in the previousSessionID field
   func test_generateNewSessionID_rotatesPreviousID() throws {
-    identifiers.generateNewSessionID()
+    let firstSessionInfo = generator.generateNewSession()
 
-    let firstSessionID = identifiers.sessionID
-    XCTAssert(isValidSessionID(identifiers.sessionID))
-    XCTAssertNil(identifiers.previousSessionID)
+    let firstSessionID = firstSessionInfo.sessionId
+    XCTAssert(isValidSessionID(firstSessionInfo.sessionId))
+    XCTAssertNil(firstSessionInfo.previousSessionId)
 
-    identifiers.generateNewSessionID()
+    let secondSessionInfo = generator.generateNewSession()
 
-    XCTAssert(isValidSessionID(identifiers.sessionID))
-    XCTAssert(isValidSessionID(identifiers.previousSessionID!))
+    XCTAssert(isValidSessionID(secondSessionInfo.sessionId))
+    XCTAssert(isValidSessionID(secondSessionInfo.previousSessionId!))
 
     // Ensure the new lastSessionID is equal to the sessionID from earlier
-    XCTAssertEqual(identifiers.previousSessionID, firstSessionID)
+    XCTAssertEqual(secondSessionInfo.previousSessionId, firstSessionID)
```

Comment:
Can we add tests around sampling?

Reply:
Added a couple of tests already. But somehow missed checking them in. Staged them now.